ui/admin/app/components/form/credential/json/secret-editor/index.js

```diff
@@ -0,0 +1,25 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class FormCredentialJsonSecretEditorComponent extends Component {
+  // =actions
+  @tracked allowEditing = false;
+
+  @action
+  allowEdit() {
+    this.allowEditing = true;
+  }
+
+  get isDetailView() {
+    return !this.args.isNew && !this.args.isEditing;
+  }
+
+  get isEditView() {
+    return !this.args.isNew && this.args.isEditing && !this.allowEditing;
```

Comment:
So it is an edit view if it's not new and in editing mode and can't edit?

Reply:
Yes, this is confusing a bit. I will add comments here to give more context. To answer your question though, yes.
When editing a JSON credential, the secret editor can have 2 states. It can be in the edit state for making changes to the name and description. This keeps the `json_object` value an empty string. This is important for when the data is serialized. Since `json_object` is marked as a secret, it will remove that attribute from the API request. 
The second state is editing everything. This will "activate" the editor for the user and they can change the secret at this point.
Please let me know if you have more questions. I will add those comments now.

Reply:
Ty for clarifying. I think comments and renaming to an elaborate variable name might help!